### ISSUE TITLE
Left UI Shell: The Focus Order fails

### ISSUE THREAD

**Issue by tay1orjones** [carbon-design-system/carbon#13416]
Originally outlined in the parent issue: 

> Problem 1: The Focus Order fails
> When the hamburger menu is activated, it discloses a left nav panel.
> The traditional and expected behaviour is that the user's next navigation point will be the first item in this revealed panel, either via tab (if it's just revealing a list of links, etc) or via down arrow (if it is revealing a menu).
> Instead, the focus progresses across the top navigation, and does not reach the first item in the left nav until all items in the top nav have been reached.
> It appears that the same tab order established when the top and left nav were persistent, has not been altered. This seems to constitute a failure of [Focus Order](https://www.w3.org/WAI/WCAG21/Understanding/focus-order.html). Combined with other issues to be detailed, this leads to a higher severity result for keyboard users.

**Comment by tay1orjones** [comment]
> The traditional and expected behaviour is that the user's next navigation point will be the first item in this revealed panel, either via tab (if it's just revealing a list of links, etc) or via down arrow (if it is revealing a menu).

I believe this is wrong based on the guidance from the [disclosure navigation](https://www.w3.org/WAI/ARIA/apg/patterns/disclosure/examples/disclosure-navigation/) pattern

> 5. Optional navigation keys (Arrow keys, Home, and End):
> ...
>    3. If implemented, the optional navigation keys supplement, but do not replace, tabbing among buttons and links. This is because the buttons and links are not contained by an element with a widget role, such as grid, that is expected to occupy only one stop in the page Tab sequence and manage focus for all its descendants.

Navigation via arrow keys would be an optional enhancement that I don't think we could easily achieve at this time based on how the UIShell components are composed and structured.
